diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,255 @@
+                nodeAfterKdoc.leaveOnlyOneNewLine()
+            }
+        }
+    }
+
+    private fun handleKdocComments(node: ASTNode) {
+        if (node.treeParent.treeParent.elementType == BLOCK) {
+            checkBlockComments(node.treeParent) // node.treeParent is a node that contains a comment.
+        } else if (node.treeParent.elementType != IF){
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleEolComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
+            checkBlockComments(node)
+        } else if (node.treeParent.lastChildNode != node && node.treeParent.elementType != IF) {
+            checkClassComment(node)
+        }
+
+        checkWhiteSpaceBeforeComment(node)
+    }
+
+    private fun handleBlockComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
+            checkBlockComments(node)
+        } else if (node.treeParent.lastChildNode != node && node.treeParent.elementType != IF) {
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleIfElse(node: ASTNode) {
+        if(node.hasChildOfType(ELSE)) {
+            val elseKeyWord = node.getFirstChildWithType(ELSE_KEYWORD)!!
+            if(elseKeyWord.treePrev.treePrev.elementType == EOL_COMMENT ||
+                    elseKeyWord.treePrev.treePrev.elementType == KDOC ||
+                    elseKeyWord.treePrev.treePrev.elementType == BLOCK_COMMENT) {
+                IF_ELSE_COMMENTS.warnAndFix(configRules, emitWarn, isFixMode, elseKeyWord.treePrev.treePrev.text, node.startOffset) {
+                    val elseBlock = node.getFirstChildWithType(ELSE)!!
+                    if(elseBlock.hasChildOfType(BLOCK)) {
+                        elseBlock.getFirstChildWithType(BLOCK)!!.addChild(elseKeyWord.treePrev.treePrev,
+                                elseBlock.getFirstChildWithType(BLOCK)!!.firstChildNode.treeNext)
+                        elseBlock.getFirstChildWithType(BLOCK)!!.addChild(PsiWhiteSpaceImpl("\n"),
+                                elseBlock.getFirstChildWithType(BLOCK)!!.firstChildNode.treeNext)
+                        node.removeChild(elseKeyWord.treePrev.treePrev)
+                    } else {
+                        val text = "else { \n${elseBlock.treePrev.treePrev.text}\n ${elseBlock.text} \n }"
+                        node.removeChild(elseBlock)
+                        node.addChild(KotlinParser().createNode(text), null)
+                    }
+
+                }
+            }
+        }
+    }
+
+    private fun checkBlockComments(node: ASTNode) {
```

Comment:
'Block comment' usually means `/* ... */`. This method should be called `checkCommentInBlock`. BTW, it seems that it handles not only blocks, so even this name is inappropriate.